pytorch_lightning/trainer/supporters.py

```diff
@@ -457,6 +464,19 @@ def _wrap_loaders_max_size_cycle(self) -> Any:
             )
             state.reset()
 
+    def _apply_cycle_iterator_length(self):
+        if self.mode == "max_size_cycle":
```

Comment:
```suggestion
        if self.mode != "max_size_cycle":
            return
```
invert the if statement?

Reply:
Done !